compiler/rustc_metadata/src/rmeta/decoder.rs

```diff
@@ -1265,6 +1265,39 @@ impl<'a, 'tcx> CrateMetadataRef<'a> {
         })
     }
 
+    fn get_ambiguity_mod_child(self, id: DefIndex, sess: &Session) -> AmbiguityModChild {
+        let ident = self.item_ident(id, sess);
+        let res = Res::Def(self.def_kind(id), self.local_def_id(id));
+        let vis = self.get_visibility(id);
+
+        AmbiguityModChild { ident, res, vis, reexport_chain: Default::default() }
+    }
+
+    fn get_ambiguity_module_children(
+        self,
+        id: DefIndex,
+        sess: &'a Session,
+    ) -> impl Iterator<Item = AmbiguityModChild> + 'a {
+        iter::from_coroutine(move || {
+            if self.root.proc_macro_data.is_none() {
```

Comment:
Redundant condition, ambiguity children lists are empty for proc macro crates anyway.